src/System.Text.RegularExpressions/tests/RegexGroupCollectionTests.cs

```diff
@@ -0,0 +1,18 @@
+﻿using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using Xunit;
+
+public class RegexGroupCollectionTests
+{
+    [Fact]
+    public static void GetGroupTest()
+    {
+        var caps = new Dictionary<int, int>
+        {
+            { 0, 0 }
+        };
+        var groups = new GroupCollection(null, caps);
```

Comment:
Instead of directly creating this internal collection I would much prefer you actually write a test that actually the code would hit from the public API surface. As I stated in general we try to avoid exposing our internals unless there is good reason.